src/trigger.rs

```diff
@@ -0,0 +1,15 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
I find it a bit weird to have a module with just one trait. How about leaving it to src/lib.rs for now? In the future we can move it to a specific module (for example one with `events` if we find such a module useful). It is subjective, but I think the overhead of using it from a separate module defeats the separation purpose.

Reply:
No strong preference, I moved it to `lib.rs`.